lr-cli/cortx_setup/commands/cluster/config/component.py

```diff
@@ -70,20 +69,43 @@ def run(self, component_group):
             cortx_components = get_cortx_states()
 
             if component_group is None:
-                for component_group in noncortx_components:
+                self.logger.debug(f"Deploying prerequisites components on nodes")
+                self._configure(
+                    noncortx_components['prerequisites']
+                )
+                for component_group in cortx_components:
                     self.logger.debug(f"Deploying {component_group} components on nodes")
                     self._configure(
-                        noncortx_components[component_group]
+                        cortx_components[component_group],
+                        stages=['config.config', 'config.init_mod']
                     )
+                self.logger.debug(f"Deploying ha components on nodes")
+                self._configure(
+                    ['ha.cortx-ha'],
+                    stages=[
+                        'config.post_install','config.prepare',
+                        'config.config', 'config.init_mod']
+                )
+            elif component_group in noncortx_components:
+                self.logger.debug(f"Deploying prerequisites components on nodes")
+                self._configure(
+                    noncortx_components['prerequisites']
+                )                        
             elif component_group in cortx_components:
                 self.logger.debug(f"Deploying {component_group} components on nodes")
                 self._configure(
                     cortx_components[component_group],
                     stages=['config.config', 'config.init_mod']
                 )
-
-            self.logger.info(f"Deployment done for {component_group} components")
-
+            elif component_group == 'ha':
+                self.logger.debug(f"Deploying prerequisites components on nodes")
+                self._configure(
+                    ['ha.cortx-ha'],
+                    stages=[
+                        'config.post_install','config.prepare',
+                        'config.config', 'config.init_mod']
```

Comment:
Instead of config.init_mod did we ever supported config.init
Why we had init_mod in first place?

Reply:
config.init is already part of our salt states logic , so to avoid confusion we have created config.init_mod to call init provisioner of components